go-controller/pkg/util/multi_network.go

```diff
@@ -939,6 +939,13 @@ func GetPodNADToNetworkMappingWithActiveNetwork(pod *kapi.Pod, nInfo NetInfo, ac
 		Name:      activeNetworkNADKey[1],
 	}
 
+	if nInfo.IsPrimaryNetwork() && nInfo.TopologyType() == types.Layer2Topology && nInfo.AllowsPersistentIPs() {
+		ipamClaimName, wasPersistentIPRequested := pod.Annotations[OvnUDNIPAMClaimName]
+		if wasPersistentIPRequested {
+			networkSelections[activeNetworkNADs[0]].IPAMClaimReference = ipamClaimName
+		}
```

Comment:
nit: wouldn't just this be equivalent:
```
networkSelections[activeNetworkNADs[0]].IPAMClaimReference = pod.Annotations[OvnUDNIPAMClaimName]
```

Reply:
maybe. I try to repeat what I think are best practices without thinking too much. 
Always checking if something was found seems good. 

In this particular case I think you're right.